llvm/test/CodeGen/RISCV/rvv/fixed-vectors-mask-buildvec.ll

```diff
@@ -140,8 +140,9 @@ define <4 x i1> @buildvec_mask_v4i1() {
 define <4 x i1> @buildvec_mask_nonconst_v4i1(i1 %x, i1 %y) {
 ; CHECK-LABEL: buildvec_mask_nonconst_v4i1:
 ; CHECK:       # %bb.0:
-; CHECK-NEXT:    vsetivli zero, 4, e8, mf4, ta, ma
+; CHECK-NEXT:    vsetivli zero, 1, e8, mf8, ta, ma
```

Comment:
I think this one is most likely fallout from the Init Undef interaction I flagged on the prior patch (moving post phi elimination).

Reply:
There are different order of `PseudoVMV_V_X_MF4` and `PseudoVMV_V_I_MF8` between post-ra nad pre-ra in vsetvl 
insertion pass.

```
; before pre-ra vsetvl insertion
    %2:vr = PseudoVMV_V_X_MF4 undef %2, %1, 4, 3 /* e8 */, 0 /* tu, mu */, implicit $vl, implicit $vtype
    %6:vr = PseudoVMV_V_I_MF8 undef %6, 3, 1, 3 /* e8 */, 0 /* tu, mu */, implicit $vl, implicit $vtype
```

```
; before post-ra vsetvl insertion
    renamable $v0 = PseudoVMV_V_I_MF8 undef renamable $v0, 3, 1, 3 /* e8 */, 0 /* tu, mu */
    renamable $v8 = PseudoVMV_V_X_MF4 undef renamable $v8, %1, 4, 3 /* e8 */, 0 /* tu, mu */
```

> machine scheduler will make them in same order (`VMV_V_I` then `VMV_V_X`). 

For `emitVSETVLIs`, it needs to insert an extra VSETVL with `VMV_V_I` then `VMV_V_X` order. @lukel97, is there a chance to fix this during the `RISCVCoalesceVSETVLI`?

Reply:
Yeah I was just looking into this. The issue lies with a discrepancy between getDemanded() (used by RISCVCoalesceVSETVLI) and needVSETVLI (used by RISCVInsertVSETVLI). We have rules for relaxing the demanded fields for slides and vmv.v.i/vmv.v.x in the latter but not the former. 

I think we can unify these. The primary reason why they aren't currently is because these rules check the VL operand, and there's a comment in getDemanded() that warns against this:

```
  // Warning: This function has to work on both the lowered (i.e. post
  // emitVSETVLIs) and pre-lowering forms.  The main implication of this is
  // that it can't use the value of a SEW, VL, or Policy operand as they might
  // be stale after lowering.
```

But I don't think this actually applies since we only care about what is demanded for **original** instruction, not what the instruction may currently compute with relaxed vsetvlis. 

I'll try and see if I can relax this. (These rules also relax LMUL, which requires knowledge of what the new LMUL is going to be so we don't accidentally increase it to something that will clobber. May need to shuffle things about.)

Reply:
See #92513 for the last bit of my comment about shuffling things about